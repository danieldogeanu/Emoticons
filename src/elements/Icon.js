import React from 'react';

const Icon = (props) => (
	<svg className={'icon ' + props.name}>
		<use xlinkHref={'#' + props.name} />
	</svg>
);

export default Icon;
