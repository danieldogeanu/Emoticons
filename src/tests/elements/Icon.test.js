import React from 'react';
import {render} from '@testing-library/react';
import Icon from '../../elements/Icon';

describe('Icon Element', () => {

	it('renders svg icon properly', () => {
		const testIcon = 'test';
		const {container} = render(<Icon name={testIcon} />);
		const renderedIcon = container.querySelector('svg');
		const useElement = container.querySelector('use');

		expect(renderedIcon).toBeInTheDocument();
		expect(renderedIcon).toHaveClass('icon', testIcon);
		expect(renderedIcon).toContainElement(useElement);
		expect(renderedIcon).toMatchSnapshot();
		expect(useElement).toHaveAttribute('xlink:href', `#${testIcon}`);
	});

});
