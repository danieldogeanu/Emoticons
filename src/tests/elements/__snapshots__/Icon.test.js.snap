// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Element renders svg icon properly 1`] = `
<svg
  class="icon test"
>
  <use
    xlink:href="#test"
  />
</svg>
`;
